assets/js/admin/course-general-sidebar.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect, subscribe, select } from '@wordpress/data';
+import {
+	PanelBody,
+	CheckboxControl,
+	SelectControl,
+	HorizontalRule,
+} from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+import apiFetch from '@wordpress/api-fetch';
+import { useState } from '@wordpress/element';
+
+const CourseGeneralSidebar = () => {
+	const course = useSelect( ( select ) => {
+		return select( 'core/editor' ).getCurrentPost();
+	} );
+	const [ author, setAuthor ] = useState( course.author );
+
+	let courses = window.courseSettingsSidebar.courses;
+	if ( courses && courses.length ) {
+		courses = courses.map( ( course ) => {
+			return { label: course.post_title, value: course.ID };
+		} );
+		courses.push( { label: __( 'None', 'sensei-lms' ), value: 0 } );
+	}
+
+	let teachers = window.courseSettingsSidebar.teachers;
+	if ( teachers && teachers.length ) {
+		teachers = teachers.map( ( usr ) => {
+			return { label: usr.display_name, value: usr.id };
+		} );
+	}
+
+	const [ meta, setMeta ] = useEntityProp( 'postType', 'course', 'meta' );
+	const featured = meta._course_featured;
+	const prerequisite = meta._course_prerequisite;
+	const notification = meta.disable_notification;
+
+	const unsubscribe = subscribe( () => {
+		let editor = select( 'core/editor' );
```

Comment:
There is a helper, [`editorLifecycle`](https://github.com/Automattic/sensei/blob/7c7a1c4a1244b5c5c11538dce4e38d44e28d18ae/assets/shared/helpers/editor-lifecycle.js#L17) that could work here. Also I think this should be in an effect and not done in every render. `editorLifecycle` will take care of running it after every save. Something like:

```
	useEffect( () => editorLifecycle( {
		onSave: () => {
			apiFetch( ... )
		},
	} ) );
```